rewards-distribution/ethereum/test/rewards_distribution_it.js

```diff
@@ -0,0 +1,718 @@
+
+            await d.announceDistributionEvent(distributionEvent);
+            await d.executeBatch(distributionEvent, 0);
+        });
+
+        it("fails if batch or was not announced at exact position, but succeeds for declared batches", async () => {
+            const d = await Driver.newWithContracts(owner);
+
+            await d.assignTokenToContract(d.getBatchAmount(0));
+
+            await d.announceDistributionEvent(distributionEvent);
+
+            await expectRevert(d.executeBatch(distributionEvent, 0, d.batches[0].slice(1))); // wrong batch list
+
+            await expectRevert(d.executeBatch(distributionEvent, 1, d.batches[0])); // wrong batchIndex
+
+            await d.executeBatch(distributionEvent, 0, d.batches[0]);
+        });
+
+        it("fails if batch array lengths dont match", async () => {
+            const d = await Driver.newWithContracts(owner);
+
+            await d.assignTokenToContract(d.getBatchAmount(0));
+
+            await d.announceDistributionEvent(distributionEvent);
+
+            const batch = d.batches[0];
+            const error = await expectRevert(d.rewards.executeCommittedBatch(distributionEvent, batch.map(r => r.address), batch.slice(1).map(r => r.amount), 0));
+            expect(error).to.have.property('reason', 'array length mismatch')
+        });
+
+        it("fails if no recipients in batch zero recipient", async () => {
+            const d = await Driver.newWithContracts(owner);
+
+            // TODO - instead of violating encapsulation use builder pattern to replace batch
+
+            // set the first address in the batch to 0x0 and correct the hash
+            d.batches[0] = [];
+            d.batchHashes[0] = RewardsClient.hashBatch(0, d.batches[0]);
+
+            await d.assignTokenToContract(d.getBatchAmount(0));
+
+            await d.announceDistributionEvent(distributionEvent);
+
+            const error = await expectRevert(d.executeBatch(distributionEvent, 0));
+            expect(error).to.have.property("reason", "at least one reward must be included in a batch");
+        });
+
+        it("fails for zero recipient", async () => {
+            const d = await Driver.newWithContracts(owner);
+
+            // TODO - instead of violating encapsulation use builder pattern to replace batch
+            const firstBatch = d.batches[0];
+
+            // set the first address in the batch to 0x0 and correct the hash
+            firstBatch[0].address = firstBatch[0].address.replace(/[^x]/g, "0");
+            d.batchHashes[0] = RewardsClient.hashBatch(0, firstBatch);
+
+            await d.assignTokenToContract(d.getBatchAmount(0));
+
+            await d.announceDistributionEvent(distributionEvent);
+
+            const error = await expectRevert(d.executeBatch(distributionEvent, 0));
+            expect(error).to.have.property("reason", "recipient must be a valid address");
+        });
+
+        it("succeeds for zero amount", async () => {
```

Comment:
why is this a requirement?

Reply:
to allow explicitly indicating that nothing was rewarded

Reply:
why?